### PR TITLE
Add ProcessMemoryMetrics ProcessThreadMetrics

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
|  Name | Decription   | Graph  |  Line   | 
|---|---|---|---|
|process.memory.vss   | Virtual set size. The amount of virtual memory the process can access. Mostly useles, but included for completeness sake.  | N/a  | N/a  |
| process.memory.rss  | Resident set size. The amount of process memory currently in RAM.   | 1  |  rss  |
|process.memory.pss   | Proportional set size. The amount of process memory currently in RAM, accounting for shared pages among processes. This metric is the most accurate in terms of "real" memory usage.  | 1  | pss  |
|process.memory.swap   |The amount of process memory paged out to swap.   | 1   | swap  |
|process.memory.swappss   | The amount of process memory paged out to swap accounting for shared memory among processes. Since Linux 4.3. Will return -1 if your kernel is older. As with pss, the most accurate metric to watch.  | 1  |  swappss |
| process.threads  | The number of process threads as seen by the operating system.  | 3  |  process |
| process.memory.pss +process.memory.swap | Total memory used by JVM | 1  |  phys |
| process.memory.pss +process.memory.swap - jvm_memory_committed_bytes | Off-heap memory used by JVM | 2  |  native |

NEW

NOW

 CQ micrometer-jvm-extras:0.1.3  https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19016 

### What issues does this PR fix or reference?
n/a

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

**Comment by skabashnyuk** [comment]
CC @ibuziuk @rhopp

**Comment by skabashnyuk** [comment]
ci-test

**Review by metlos — APPROVED** [review]

**Review by garagatyi — APPROVED** [review]

**Comment by skabashnyuk** [comment]
@eclipse/eclipse-che-qa can you take a look?

File: core/che-core-metrics-core/pom.xml [hunk 1]
```diff
@@ -21,6 +21,15 @@
     </parent>
     <artifactId>che-core-metrics-core</artifactId>
     <name>Che Core :: Metrics :: Core</name>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.github.mweirauch</groupId>
+                <artifactId>micrometer-jvm-extras</artifactId>
+                <version>0.1.3</version>
```
Comment (sleshchenko) [inline core/che-core-metrics-core/pom.xml]: Could you please add a comment here why this dependency should be declared in a dedicated `che-core-metrics-core` dependency management section but not in che parent along with other dependencies?

**Review by sleshchenko — APPROVED** [review]
LGTM

File: core/che-core-metrics-core/pom.xml (hunk 1 above)
Comment (skabashnyuk) [reply core/che-core-metrics-core/pom.xml]: I'll move this dependency to che-parent before the merge

**Comment by skabashnyuk** [comment]
ci-test

**Comment by SkorikSergey** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1579//Selenium_20tests_20report/) doesn't show any regression against this Pull Request.

**Comment by skabashnyuk** [comment]
ci-build

**Pull request merged**